javascript/ql/lib/semmle/javascript/ApiGraphs.qll

```diff
@@ -210,7 +210,12 @@ module API {
      * This is similar to `asSink()` but additionally includes nodes that transitively reach a sink by data flow.
      * See `asSink()` for examples.
      */
-    DataFlow::Node getAValueReachingSink() { result = Impl::trackDefNode(this.asSink()) }
+    DataFlow::Node getAValueReachingSink() {
+      result = Impl::trackDefNode(this.asSink())
+      or
+      // in case `asSink()` is not a `SourceNode`.
+      result = this.asSink()
+    }
```

Comment:
If `asSink()` is not a `SourceNode`, you should simply get a local source for it, is that not good enough? I could se it failing, if `rhs(_, _, asSink())` does not hold at all, but then that predicate should perhaps be extended? What is the concrete value that prompted this?

Reply:
> If asSink() is not a `SourceNode`, you should simply get a local source for it 

The problem is that `.getALocalSource()` has no result.  
A boolean literal is not a `SourceNode`, and `.getALocalSource()` has no result.   
I think `SourceNode` in Python is a bit different. I suspect that you are guaranteed that `.getALocalSource()` has a result? 

--- 

The motivation was a boolean literal being written to a property.  

---- 

Now that I think about it: `getAValueReachingSink` adds nothing when we are tracking booleans, I would need a small-step type-backtracker (because that doesn't depend on `SourceNode`).

Reply:
Indeed, in Python there will always be a local source. If nothing else, then the node itself.